imports/ui/pages/AuthPageSignIn.jsx

```diff
@@ -1,111 +1,105 @@
+      newErrors.email = i18n.__('pages.authPageSignIn.emailRequired');
     }
     if (!password) {
-      errors.password = i18n.__('pages.authPageSignIn.passwordRequired');
+      newErrors.password = i18n.__('pages.authPageSignIn.passwordRequired');
     }
 
-    this.setState({ errors });
-    if (Object.keys(errors).length) {
+    setErrors(newErrors);
+    if (Object.keys(newErrors).length) {
       return;
     }
 
     Meteor.loginWithPassword(email, password, (err) => {
       if (err) {
-        this.setState({
-          errors: { none: err.reason },
-        });
+        if (err) {
+          setErrors({ none: err.reason });
+        }
       } else {
-        this.redirectTo('/');
+        setRedirectTo('/');
       }
     });
-  }
+  };
 
-  render() {
-    const { errors } = this.state;
-    const errorMessages = Object.keys(errors).map(key => errors[key]);
-    const errorClass = key => errors[key] && 'error';
+  const errorMessages = Object.values(errors);
+  const errorClass = key => errors[key] && 'error';
 
-    const content = (
-      <div className="wrapper-auth">
-        <h1 className="title-auth">
-          {i18n.__('pages.authPageSignIn.signIn')}
-        </h1>
-        <p className="subtitle-auth">
-          {i18n.__('pages.authPageSignIn.signInReason')}
-        </p>
-        <form onSubmit={this.onSubmit}>
-          <div className="list-errors">
-            {errorMessages.map(msg => (
-              <div className="list-item" key={msg}>{msg}</div>
-            ))}
-          </div>
-          <div className={`input-symbol ${errorClass('email')}`}>
-            <input
-              type="email"
-              name="email"
-              ref={(c) => { this.email = c; }}
-              placeholder={i18n.__('pages.authPageSignIn.yourEmail')}
-            />
-            <span
-              className="icon-email"
-              title={i18n.__('pages.authPageSignIn.yourEmail')}
-            />
-          </div>
-          <div className={`input-symbol ${errorClass('password')}`}>
-            <input
-              type="password"
-              name="password"
-              ref={(c) => { this.password = c; }}
-              placeholder={i18n.__('pages.authPageSignIn.password')}
-            />
-            <span
-              className="icon-lock"
-              title={i18n.__('pages.authPageSignIn.password')}
-            />
-          </div>
-          <button type="submit" className="btn-primary">
-            {i18n.__('pages.authPageSignIn.signInButton')}
-          </button>
-        </form>
-      </div>
-    );
+  const link = (
+    <Link to="/join" className="link-auth-alt">
+      {i18n.__('pages.authPageSignIn.needAccount')}
+    </Link>
+  );
 
-    const link = (
-      <Link to="/join" className="link-auth-alt">
-        {i18n.__('pages.authPageSignIn.needAccount')}
-      </Link>
+  return redirectTo
+    ? <Redirect to={redirectTo} />
+    : (
```

Comment:
Using useHistory was also my first though. I was surprised by the former architecture and wondered why history wasn't used directly using a HOC. Then I though that it might be for security reasons, so I tried to keep the architecture. I agree to change, that would be much simple and clear.